contracts/account/extensions/AccountERC7579.sol

```diff
@@ -0,0 +1,370 @@
+     * Requirements:
+     *
+     * * Module type must be supported. See {supportsModule}. Reverts with {ERC7579UnsupportedModuleType}.
+     * * Module must be of the given type. Reverts with {ERC7579MismatchedModuleTypeId}.
+     * * Module must not be already installed. Reverts with {ERC7579AlreadyInstalledModule}.
+     *
+     * Emits a {ModuleInstalled} event.
+     */
+    function _installModule(uint256 moduleTypeId, address module, bytes memory initData) internal virtual {
+        require(supportsModule(moduleTypeId), ERC7579Utils.ERC7579UnsupportedModuleType(moduleTypeId));
+        require(
+            IERC7579Module(module).isModuleType(moduleTypeId),
+            ERC7579Utils.ERC7579MismatchedModuleTypeId(moduleTypeId, module)
+        );
+
+        bool exists;
+        if (moduleTypeId == MODULE_TYPE_VALIDATOR) exists = !_validators.add(module);
+        if (moduleTypeId == MODULE_TYPE_EXECUTOR) exists = !_executors.add(module);
+        require(!exists, ERC7579Utils.ERC7579AlreadyInstalledModule(moduleTypeId, module));
+
+        if (moduleTypeId == MODULE_TYPE_FALLBACK) {
+            bytes4 selector;
+            (selector, initData) = abi.decode(initData, (bytes4, bytes));
+            require(
+                _installFallback(module, selector),
+                ERC7579Utils.ERC7579AlreadyInstalledModule(moduleTypeId, module)
+            );
+        }
+
+        IERC7579Module(module).onInstall(initData);
+        emit ModuleInstalled(moduleTypeId, module);
+    }
+
+    /**
+     * @dev Uninstalls a module of the given type with the given de-initialization data.
+     *
+     * For the fallback module type, the `deInitData` is expected to be a tuple of a 4-byte selector and the
+     * rest of the data to be sent to the handler when calling {IERC7579Module-onUninstall}.
+     *
+     * Requirements:
+     *
+     * * Module must be already installed. Reverts with {ERC7579UninstalledModule} otherwise.
+     */
+    function _uninstallModule(uint256 moduleTypeId, address module, bytes memory deInitData) internal virtual {
+        require(
+            (moduleTypeId != MODULE_TYPE_VALIDATOR || _validators.remove(module)) &&
+                (moduleTypeId != MODULE_TYPE_EXECUTOR || _executors.remove(module)),
+            ERC7579Utils.ERC7579UninstalledModule(moduleTypeId, module)
+        );
+
+        if (moduleTypeId == MODULE_TYPE_FALLBACK) {
+            bytes4 selector;
+            (selector, deInitData) = abi.decode(deInitData, (bytes4, bytes));
+            require(_uninstallFallback(module, selector), ERC7579Utils.ERC7579UninstalledModule(moduleTypeId, module));
+        }
+
+        IERC7579Module(module).onUninstall(deInitData);
+        emit ModuleUninstalled(moduleTypeId, module);
+    }
+
+    /**
+     * @dev Installs a fallback handler for the given selector. Returns true if the handler was installed,
+     */
+    function _installFallback(address module, bytes4 selector) internal virtual returns (bool) {
```

Comment:
_installModule is already internal, and can be overriden. I'd make that one private so that fallbacks are not modified without an event (and the onInstall/onUninstall call being performed)